### PR TITLE
Respect CLI logger flag precedence over KS_LOGGER

### PR DESCRIPTION
**Pull request by aaa-aashna**
## Overview

Fix `KS_LOGGER` precedence handling in `initLoggerLevel`.

Previously:

* `KS_LOGGER` was ignored when the logger remained at the default `info` level
* `KS_LOGGER` could override an explicitly provided `-l/--logger` flag

This change updates the behavior so:

* `KS_LOGGER` is applied only when the logger is still at the default level
* explicit CLI logger flags always take precedence over the environment variable

Also adds focused unit tests covering both precedence cases.

## How to Test

Run:

```bash id="0f94"
go test -count=1 ./cmd -run TestInitLoggerLevel_KSLoggerPrecedence -v
```

Optionally run the full cmd package tests:

```bash id="0f95"
go test ./cmd -v
```

## Checklist before requesting a review

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **New Features**
  * Logger configuration now respects explicit CLI logger settings and environment variable behavior across commands.

* **Bug Fixes**
  * Corrected related-resource ID handling so control rule results return the expected resource lists.

* **Tests**
  * Added and expanded tests covering logger precedence and related-resource mapping.

[Review Change Stack](https://app.coderabbit.ai/change-stack/kubescape/kubescape/pull/2123)

### CODE REVIEW

File: cmd/rootutils.go [hunk 1]
```diff
@@ -35,8 +35,9 @@ func initLogger() {
 
 func initLoggerLevel() {
 	if rootInfo.Logger == helpers.InfoLevel.String() {
-	} else if l := os.Getenv("KS_LOGGER"); l != "" {
-		rootInfo.Logger = l
+		if l := os.Getenv("KS_LOGGER"); l != "" {
```
Comment (matthyx) [inline cmd/rootutils.go:51]: This still misses one explicit-flag case: `rootInfo.Logger == "info"` cannot tell whether we are still on the default or the user explicitly passed `--logger info`. With the current code, `KS_LOGGER=debug kubescape --logger info version` still ends up at `debug`, so an explicit CLI flag does **not** always win over the environment variable as described in the PR. Can we gate the env override on whether the `logger` flag was actually changed (for example via `cmd.Flags().Changed("logger")`) and add a regression test for the explicit `--logger info` path?

File: httphandler/storage/apiserver.go [hunk 2]
```diff
@@ -486,8 +486,7 @@ func parseScannedControlRules(control *resourcesresults.ResourceAssociatedContro
 
 		controlConfigurations := ruleToControlConfigurations(rule)
 
-		relatedResourceIds := []string{}
-		copy(relatedResourceIds, rule.RelatedResourcesIDs)
+		relatedResourceIds := append([]string(nil), rule.RelatedResourcesIDs...)
```
Comment (matthyx) [inline httphandler/storage/apiserver.go]: This copy fix works for non-empty slices, but it changes the empty/nil case from `[]` to `nil`. `ScannedControlRule.RelatedResourcesIDs` is serialized as `json:"relatedResourcesIDs"` without `omitempty`, so callers will now see `null` instead of `[]`. I reproduced that with a focused test against `parseScannedControlRules`. Can we preserve the previous empty-slice behavior while copying (for example `make([]string, len(rule.RelatedResourcesIDs)); copy(...)`) and add a regression test for the empty/nil case?

**Review by matthyx — CHANGES_REQUESTED** [review]
Left two blocker comments:

1. `KS_LOGGER` still overrides an explicit `--logger info` because the code keys off the final value instead of whether the flag was actually set.
2. The `RelatedResourcesIDs` copy changes empty slices to `nil`, which changes the storage API JSON shape from `[]` to `null`.

Once those are fixed, the rest of the change looks good.

**Comment by aaa-aashna** [comment]
@matthyx 
Addressed both review comments:

* updated logger precedence handling to use Cobra flag `Changed` state so explicitly provided `--logger` values always win over `KS_LOGGER`
* preserved empty slice behavior for `RelatedResourcesIDs` while still fixing the copy issue

Also updated the related test coverage accordingly.

**Review by matthyx — COMMENTED** [review]
Rechecked the latest head. The `RelatedResourcesIDs` fix looks good now, but the logger precedence issue is still open: `KS_LOGGER=debug kubescape --logger info version` still ends up at `debug`, so an explicit CLI flag still does not always win over the environment variable. I did not find any new blockers beyond that remaining logger case.

**Comment by aaa-aashna** [comment]
@matthyx 
Thanks for catching that case.

Updated the logic to check the root command’s logger flag state for inherited persistent flags, and added regression coverage for:

```bash id="0f178"
KS_LOGGER=debug kubescape --logger info version
```

so explicit `--logger` values now correctly win over `KS_LOGGER` on subcommands as well.

**Review by matthyx — APPROVED** [review]
Rechecked the latest head. The logger precedence fix now correctly respects an explicit `--logger info`, the relatedResourcesIDs change is safe, and I don’t see any blocker-level issues remaining.
